runtime/parser2/expression.go

```diff
@@ -626,6 +615,95 @@ func defineLessThanOrTypeArgumentsExpression() {
 		})
 }
 
+func defineGreaterThanOrBitwiseRightShiftExpression() {
+
+	setExprMetaLeftDenotation(
+		lexer.TokenGreater,
+		func(p *parser, rightBindingPower int, left ast.Expression) (result ast.Expression, done bool) {
+
+			// If the right binding power is higher than any of the potential cases,
+			// then return early
+
+			if rightBindingPower >= exprLeftBindingPowerBitwiseShift &&
+				rightBindingPower >= exprLeftBindingPowerComparison {
+
+				return left, true
+			}
+
+			// Start buffering before skipping the `>` token,
+			// so it can be replayed in case the right binding power
+			// was higher than the determined left binding power.
+
+			p.startBuffering()
+
+			// Skip the `>` token.
+			p.next()
+
+			// If another '>' token appears immediately,
+			// then the operator is actually a bitwise right shift operator
+
+			isBitwiseShift := p.current.Is(lexer.TokenGreater)
+
+			var operation ast.Operation
+			var nextRightBindingPower int
+
+			if isBitwiseShift {
+
+				operation = ast.OperationBitwiseRightShift
+
+				// The expression was determined to be a bitwise shift.
+				// Still, it should have maybe not been parsed if the right binding power
+				// was higher. In that case, replay the buffered tokens and stop.
+
+				if rightBindingPower >= exprLeftBindingPowerBitwiseShift {
```

Comment:
```suggestion
				if rightBindingPower >= exprLeftBindingPowerBitwiseShift {
				// for instance, this could happen when parsing `a<T<U>>()`
				// where `()`'s binding power is higher than `>>`.
```